lib/cadet/achievements/achievements.ex

```diff
@@ -0,0 +1,238 @@
+defmodule Cadet.Achievements do
+  @moduledoc """
+  The Achievement entity stores metadata of a students' assessment
+  """
+  use Cadet, [:context, :display]
+
+  alias Cadet.Achievements.{Achievement, AchievementGoal}
+  alias Cadet.Accounts.User
+
+  import Ecto.Query
+
+  # Gets all achieveemnts of a particular user
+  def all_achievements(user) do
+    achievements =
+      Achievement
+      |> order_by([a], [a.inferencer_id])
+      |> Repo.all()
+      |> Repo.preload(goals: [:achievement, :user])
```

Comment:
This preload will be unnecessary if you use a join to load the necessary goals only.

N.B. This line currently loads *all* the goal entries.. So it will load, let's say, 600 users * 50 achievements = 30,000 rows. :(

Reply:
ahhh i see i see alrights